roda-ui/roda-wui/src/main/java/org/roda/wui/client/portal/BrowseDIPPortal.java

```diff
@@ -0,0 +1,381 @@
+import org.roda.wui.client.common.TitlePanel;
+import org.roda.wui.client.common.UserLogin;
+import org.roda.wui.client.common.actions.DisseminationFileActions;
+import org.roda.wui.client.common.lists.DIPFileList;
+import org.roda.wui.client.common.lists.pagination.ListSelectionUtils;
+import org.roda.wui.client.common.lists.utils.AsyncTableCellOptions;
+import org.roda.wui.client.common.lists.utils.ListBuilder;
+import org.roda.wui.client.common.search.SearchWrapper;
+import org.roda.wui.client.common.utils.AsyncCallbackUtils;
+import org.roda.wui.client.common.utils.IndexedDIPUtils;
+import org.roda.wui.client.common.utils.JavascriptUtils;
+import org.roda.wui.common.client.HistoryResolver;
+import org.roda.wui.common.client.tools.HistoryUtils;
+import org.roda.wui.common.client.tools.RestErrorOverlayType;
+import org.roda.wui.common.client.tools.RestUtils;
+import org.roda.wui.common.client.tools.StringUtils;
+import org.roda.wui.common.client.widgets.HTMLWidgetWrapper;
+import org.roda.wui.common.client.widgets.Toast;
+import org.roda.wui.common.client.widgets.wcag.AccessibleFocusPanel;
+import org.roda.wui.common.client.widgets.wcag.WCAGUtilities;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.core.client.JsonUtils;
+import com.google.gwt.http.client.Request;
+import com.google.gwt.http.client.RequestBuilder;
+import com.google.gwt.http.client.RequestCallback;
+import com.google.gwt.http.client.RequestException;
+import com.google.gwt.http.client.Response;
+import com.google.gwt.i18n.client.LocaleInfo;
+import com.google.gwt.safehtml.shared.SafeHtml;
+import com.google.gwt.safehtml.shared.SafeHtmlBuilder;
+import com.google.gwt.safehtml.shared.SafeHtmlUtils;
+import com.google.gwt.safehtml.shared.SafeUri;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.History;
+import com.google.gwt.user.client.Window;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.HTML;
+import com.google.gwt.user.client.ui.SimplePanel;
+import com.google.gwt.user.client.ui.Widget;
+
+import config.i18n.client.ClientMessages;
+
+/**
+ * @author Eduardo Teixeira <eteixeira@keep.pt>
+ */
+public class BrowseDIPPortal extends Composite {
+
+  public static final HistoryResolver RESOLVER = new HistoryResolver() {
+
+    @Override
+    public String getHistoryToken() {
+      return "dip";
+    }
+
+    @Override
+    public List<String> getHistoryPath() {
+      return Arrays.asList(getHistoryToken());
+    }
+
+    @Override
+    public void isCurrentUserPermitted(AsyncCallback<Boolean> callback) {
+      UserLogin.getInstance().checkRole(this, callback);
+    }
+
+    @Override
+    public void resolve(List<String> historyTokens, AsyncCallback<Widget> callback) {
+      GWT.log("BrowseDIPPortal tokens: " + historyTokens);
```

Comment:
Remove this line, we shouldn't leave GWT logs in the final version, also review for imports that are not being used in the code.